### PR TITLE
replace validate_* with puppet4 datatypes

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -79,16 +79,14 @@
...
-  $exportable = true) {
-  validate_re($name, '^(.)+$', "Must pass name to ${module_name}[${title}]")
-  validate_re($location, '^(.)+$', "Must pass location to ${module_name}[${title}]")
-  validate_re($thumbprint, '^(.)+$', "Must pass a certificate thumbprint to ${module_name}[${title}]")
+  String $password,
+  String $location,
+  String $thumbprint,
+  String $root_store                = 'LocalMachine',
+  String $store_dir                 = 'My',
+  Stdlib::Absolutepath $scripts_dir = 'C:\temp',
```
Comment (alexjfisher) [inline manifests/init.pp]: You could be more specific here and use Stdlib::Windowspath

File: manifests/init.pp [hunk 2]
```diff
@@ -79,16 +79,14 @@
...
 #
 define sslcertificate (
-  $password,
-  $location,
-  $thumbprint,
-  $root_store    = 'LocalMachine',
-  $store_dir     = 'My',
-  $scripts_dir   = 'C:\temp',
-  $exportable = true) {
-  validate_re($name, '^(.)+$', "Must pass name to ${module_name}[${title}]")
```
Comment (alexjfisher) [inline manifests/init.pp]: I wonder what this was ever for?  How could you *not* pass a name to a defined type?

File: manifests/init.pp (hunk 1 above)
Comment (bastelfreak) [inline manifests/init.pp]: fixed

File: manifests/init.pp [hunk 3]
```diff
@@ -79,16 +79,14 @@
...
-  validate_re($location, '^(.)+$', "Must pass location to ${module_name}[${title}]")
-  validate_re($thumbprint, '^(.)+$', "Must pass a certificate thumbprint to ${module_name}[${title}]")
+  String $password,
+  String $location,
+  String $thumbprint,
```
Comment (alexjfisher) [inline manifests/init.pp]: String[1] ? (the old regexes wouldn't allow empty strings)

File: manifests/init.pp (hunk 2 above)
Comment (bastelfreak) [inline manifests/init.pp]: I'm pretty sure this is some very very legacy foo. I saw this in multiple modules in the past weeks.

File: manifests/init.pp (hunk 3 above)
Comment (bastelfreak) [inline manifests/init.pp]: done

File: manifests/init.pp (hunk 2 above)
Comment (alexjfisher) [inline manifests/init.pp]: You'd really have to go out of your way to set $name to an empty string.

**Pull request merged**